sdk/script/account-manager.ts

```diff
@@ -73,523 +77,133 @@ export class AccountManager {
+                    return app;
                 } else {
-                    if (body) {
-                        reject(<CodePushError>body);
-                    } else {
-                        reject(<CodePushError>{ message: res.text, statusCode: res.status });
-                    }
+                    return null;
                 }
             });
-        });
-    }
-
-    public addApp(appName: string): Promise<App> {
-        return Promise<App>((resolve, reject, notify) => {
-            var app = <App>{ name: appName };
-
-            var request: superagent.Request<any> = superagent.post(this._serverUrl + "/apps/");
-            this.attachCredentials(request);
-
-            request.set("Content-Type", "application/json;charset=UTF-8")
-                .send(JSON.stringify(app))
-                .end((err: any, res: superagent.Response) => {
-                    if (err) {
-                        reject(<CodePushError>{ message: this.getErrorMessage(err, res) });
-                        return;
-                    }
-
-                    if (res.ok) {
-                        var location = res.header["location"];
-                        if (location && location.lastIndexOf("/") !== -1) {
-                            app.id = location.substr(location.lastIndexOf("/") + 1);
-                            resolve(app);
-                        } else {
-                            resolve(null);
-                        }
-                    } else {
-                        var body = tryJSON(res.text);
-                        if (body) {
-                            reject(<CodePushError>body);
-                        } else {
-                            reject(<CodePushError>{ message: res.text, statusCode: res.status });
-                        }
-                    }
-                });
-        });
     }
 
     public removeApp(app: App | string): Promise<void> {
-        return Promise<void>((resolve, reject, notify) => {
-            var id: string = (typeof app === "string") ? app : app.id;
-            var request: superagent.Request<any> = superagent.del(this._serverUrl + "/apps/" + id);
-            this.attachCredentials(request);
-
-            request.end((err: any, res: superagent.Response) => {
-                if (err) {
-                    reject(<CodePushError>{ message: this.getErrorMessage(err, res) });
-                    return;
-                }
-
-                if (res.ok) {
-                    resolve(null);
-                } else {
-                    var body = tryJSON(res.text);
-                    if (body) {
-                        reject(<CodePushError>body);
-                    } else {
-                        reject(<CodePushError>{ message: res.text, statusCode: res.status });
-                    }
-                }
-            });
-        });
+        var id: string = (typeof app === `string`) ? (<string><any>app) : (<App><any>app).id;
```

Comment:
Weird. That wasn't compiling before but it does now (which makes sense). Changed it to the more logical option here